### PR TITLE
Fix folder reordering bug; integrate react beautiful dnd

### PR DESCRIPTION
**Pull request by williamtriinh**
Create rank tests

Fix vitest dependency resolution error

Update drag and drop; add unit tests for rank functions

Clean up code

Fix prompts reordering

Disable item dragging when its being edited

Fix dropping items into opened, empty folders

Fix DnDing too fast

Add rounded corners to folders

Fix insertion order for new chats

Add min height to conversation/prompt droppables

Resolve unsupported nested scroll container console messages

Revert vitest version

### CODE REVIEW

File: components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ interface Props<T> {
   toggleOpen: () => void;
   handleCreateItem: () => void;
   handleCreateFolder: () => void;
-  handleDrop: (e: any) => void;
+  // handleDrop: (e: any) => void;
```
Comment (1orZero) [inline components/Sidebar/Sidebar.tsx]: Let's remove the unused comment

File: components/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const Sidebar = <T,>({
   toggleOpen,
   handleCreateItem,
   handleCreateFolder,
-  handleDrop,
+  // handleDrop,
```
Comment (1orZero) [inline components/Sidebar/Sidebar.tsx]: Let's remove the unused comment

**Comment by 1orZero** [comment]
Besides the unused comments, everything looks good to me!

**Comment by williamtriinh** [comment]
> Besides the unused comments, everything looks good to me!

All fixed!

**Review by thejackwu — APPROVED** [review]

**Comment by thejackwu** [comment]
@williamtriinh Beautifully done!

**Pull request merged**